pkg/combustion/registry.go

```diff
@@ -30,50 +28,48 @@ var haulerManifest string
 var registryScript string
 
 func configureRegistry(ctx *image.Context) ([]string, error) {
-	if IsEmbeddedArtifactRegistryEmpty(ctx.ImageDefinition.EmbeddedArtifactRegistry) {
+	if IsEmbeddedArtifactRegistryAndKubernetesManifestsEmpty(ctx) {
 		log.AuditComponentSkipped(registryComponentName)
 		return nil, nil
 	}
 
-	err := writeHaulerManifest(ctx)
-	if err != nil {
-		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("writing hauler manifest: %w", err)
-	}
-
-	err = populateHaulerStore(ctx)
+	haulerBinaryPath := fmt.Sprintf("hauler-%s", string(ctx.ImageDefinition.Image.Arch))
+	err := copyHaulerBinary(ctx, haulerBinaryPath)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("populating hauler store: %w", err)
+		return nil, fmt.Errorf("copying hauler binary: %w", err)
 	}
 
-	err = generateRegistryTar(ctx)
+	registryScriptNameResult, err := writeRegistryScript(ctx)
 	if err != nil {
 		log.AuditComponentFailed(registryComponentName)
-		return nil, fmt.Errorf("generating hauler store tar: %w", err)
+		return nil, fmt.Errorf("writing registry script: %w", err)
 	}
 
-	haulerBinaryPath := fmt.Sprintf("hauler-%s", string(ctx.ImageDefinition.Image.Arch))
-	err = copyHaulerBinary(ctx, haulerBinaryPath)
+	registriesDir := filepath.Join(ctx.CombustionDir, "registry")
```

Comment:
The dir name should be a constant in order for us to be able to easily reuse it.